backend/rag_service.py

```
@@ -73,31 +77,34 @@ def retrieve(self, query: str, threshold: float = 0.05) -> Optional[str]:
-            score = len(intersection) / len(union)
-
-            # Boost score if title words match (weighted)
-            title_tokens = prepared['title_tokens']
-            title_match = len(query_tokens.intersection(title_tokens))
-            if title_match > 0:
-                score += 0.2  # Bonus for title match
+            # Performance Boost: Use isdisjoint() for O(min(len(A), len(B))) early exit
+            if query_tokens.isdisjoint(policy_tokens):
+                score = 0.0
+            else:
+                # Jaccard Similarity: |A ∩ B| / |A ∪ B|
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF003)